### DISCUSSION TITLE
how to copy the uploaded file to the upload folder

### DISCUSSION THREAD

**Discussion by bobwatcherx** [zauberzeug/nicegui#496] [Q&A] [answered]
I want to ask .  how to copy the resulting file that I uploaded to the uploads folder
code
```py
import shutil
from nicegui import ui

def on_upload(event):
    # Mendapatkan nama file yang diunggah
    print(x)


    file_name = event.name
    print(e)
    # Menyalin file ke direktori uploads
    src_path = event.path
    dst_path = './uploads/' + file_name
    shutil.copy(src_path, dst_path)
    
    ui.notify(f'{file_name} uploaded and saved to uploads folder')
    
ui.upload(on_upload=on_upload).classes('max-w-full')

ui.run()
```
thanks

**Comment by falkoschindler** [comment, marked as answer]
Please see https://github.com/zauberzeug/nicegui/discussions/283 for a related discussion.
There is no `path` in the upload event. So you can't copy the file but need to write a new one with the data from `event.content`.
Or you use something like the [local file picker](https://github.com/zauberzeug/nicegui/tree/main/examples/local_file_picker).

**Comment by bobwatcherx** [comment]
I'm still confused why the upload file is useless for uploading. instead use something else namely local file picker . Then what is the function of the upload file if u can't do it Upload files

**Reply by falkoschindler** [reply]
Local file picker is a special solution for when NiceGUI is running on the client's machine (where your browser runs) and needs access to the file path, e.g. for analyzing the file without loading it into memory all at once. This is not possibly by "uploading" the file via normal web tools like Quasar's uploader.

In most cases, however, uploading (i.e. transmitting the raw file data) is just fine. The original file path gets lost, but usually the server is interested in the content rather than its original location. A major benefit is that the server (NiceGUI) can run on a different machine than the client (the browser). If you, e.g., implement an online image conversion service with NiceGUI, you can conveniently use `ui.upload`. The file needs to be sent to the server anyway and you don't want to get information about the client's file system.

**Reply by rodja** [reply]
@falkoschindler Since we have now implemented the standalone mode (coming with next release 1.2.0), running the frontend on the same machine as the backend will become a more common use-case.
